plugins/inventory/proxmox.py

```diff
@@ -577,22 +581,23 @@ def _populate(self):
         for node in self._get_nodes():
             if not node.get('node'):
                 continue
-
-            self.inventory.add_host(node['node'])
-            if node['type'] == 'node':
+            if not self.exclude_nodes:
```

Comment:
Shouldn't this only be excluded if `node['type'] == 'node' and self.exclude_nodes`?
```suggestion
            if not (node['type'] == 'node' and self.exclude_nodes):
```

Reply:
The original code just plainly added the node at this step, I therefore think checking for the type is unnecessary. This is due to the proxmox api, which de

Reply:
This check was not done in the original code as well. Since all hosts are children of nodes and the following code is adding specifically a node as an inventory item, I think it is implied, that at this stage only a node can be added.

So adding this check would be valid, but would not change anything.

Reply:
I know nothing about proxmox so I don't know whether the check has any effect or not. It would be great if someone else who actually uses proxmox could comment here.

In any case it is very strange that below `node['type'] == 'node'` is checked, but not here (and in the other `if`).

Reply:
I used the original code and removed the `node['type'] == 'node'` check - it did not change nor break anything. 

Furthermore, looking through the code and the api-responses, this check is **always** True, since it uses the function `self._get_nodes()` to obtain a list of each node from  `/api2/json/nodes`. And this endpoint returns a list only consisting of nodes, see the api spec at https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes. 

For example, the json-response from my proxmox  api (which is on 8.0.4/current):
```json
{"data":[{"disk":1234,"maxdisk":1234,"type":"node","maxcpu":12,"maxmem":1234,"status":"online","mem":1234,"uptime":1234,"ssl_fingerprint":"[redacted]","cpu":1234,"node":"hypervisor","id":"node/hypervisor","level":""}]}
```

BTW: The proxmox-api definition does not even document the `type`-key at that endpoint at all.

Reply:
@Thulium-Drake since you wrote the original code (this `type` check has already been in the original PR #545 and also its very first commit), do you remember why you addd this check?

Reply:
When do you think can we proceed here?

Reply:
@Thulium-Drake another ping. Would really appreciate a look!

Reply:
Sorry for the late reply, it's been mayhem over here :) I'll have a look and edit this reply ;-)

Reply:
Hey, thanks for the update. I edited my reply (the json was unreadable). We have the same json output as far as I can see. So no difference there, even though my node *was* in a cluster once.

Regarding the current issue. Thanks for your input. I would pledge to removing this check or keeping this PR as it is now then. Because:
  
> Get all nodes and loop over it
This is what `self._get_nodes()` is doing. Calling the proxmox api and retrieving all the nodes - but only nodes - from the api. 

> Add the node to the inventory and when the item we're treating is a node, add to the nodes group
Therefore the check ` if node['type'] == 'node'` is always True, since the given api endpoint will only return a list consisting of ndoes. So every `node` from  `for node in self._get_nodes()` will be added to the nodes group. 

Please correct me if I am wrong! :)

Reply:
Sounds good to me! :-)

Thank you for this work!